### PR TITLE
updated fault counts for ADI2 to include S & D from vehicle checks

### PR DESCRIPTION
**Pull request by andrewsetterfield**
## Description

Update fault counts to include Serious and dangerous faults from vehicle checks.  This effects the debrief page (Test outcome, and displays) and the office page comments section.

## Checklist

- [x] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/providers/fault-count/cat-adi-part2/fault-count.cat-adi-part2.ts [hunk 1]
```diff
@@ -35,11 +35,12 @@ export class FaultCountADIPart2Helper {
     const seriousFaultSumOfSimpleCompetencies = Object.keys(pickBy(seriousFaults)).length;
     const controlledStopSeriousFaults = (controlledStop && controlledStop.fault === CompetencyOutcome.S) ? 1 : 0;
     const eyesightTestSeriousFaults = (eyesightTest && eyesightTest.seriousFault) ? 1 : 0;
+    const seriousFaultFromVehicleChecks = vehicleChecks.seriousFault ? 1 : 0;
 
     const result =
       seriousFaultSumOfSimpleCompetencies +
       sumManoeuvreFaults(manoeuvres, CompetencyOutcome.S) +
-      FaultCountADIPart2Helper.getVehicleChecksByOutcomeFaultCountCatADIPart2(vehicleChecks, CompetencyOutcome.S) +
```
Comment (matthew2564) [inline src/providers/fault-count/cat-adi-part2/fault-count.cat-adi-part2.ts:42]: does this method not work?

File: src/providers/fault-count/cat-adi-part2/fault-count.cat-adi-part2.ts [hunk 2]
```diff
@@ -54,11 +55,12 @@ export class FaultCountADIPart2Helper {
 
     const dangerousFaultSumOfSimpleCompetencies = Object.keys(pickBy(dangerousFaults)).length;
     const controlledStopDangerousFaults = (controlledStop && controlledStop.fault === CompetencyOutcome.D) ? 1 : 0;
+    const dangerousFaultFromVehicleChecks = vehicleChecks.dangerousFault ? 1 : 0;
```
Comment (matthew2564) [inline src/providers/fault-count/cat-adi-part2/fault-count.cat-adi-part2.ts:58]: same as above^

File: src/providers/fault-count/cat-adi-part2/fault-count.cat-adi-part2.ts [hunk 3]
```diff
@@ -54,11 +55,12 @@ export class FaultCountADIPart2Helper {
...
 
     const result =
       dangerousFaultSumOfSimpleCompetencies +
       sumManoeuvreFaults(manoeuvres, CompetencyOutcome.D) +
-      FaultCountADIPart2Helper.getVehicleChecksByOutcomeFaultCountCatADIPart2(vehicleChecks, CompetencyOutcome.D) +
```
Comment (matthew2564) [inline src/providers/fault-count/cat-adi-part2/fault-count.cat-adi-part2.ts:61]: same as above^

File: src/providers/fault-count/cat-adi-part2/fault-count.cat-adi-part2.ts (hunk 1 above)
Comment (andrewsetterfield) [reply src/providers/fault-count/cat-adi-part2/fault-count.cat-adi-part2.ts:42]: It works fine, however the data for vehicle checks dont record serious faults within the outcome, it has its own boolean in the schema.

File: src/providers/fault-count/cat-adi-part2/fault-count.cat-adi-part2.ts (hunk 3 above)
Comment (andrewsetterfield) [reply src/providers/fault-count/cat-adi-part2/fault-count.cat-adi-part2.ts:61]: same as above^

File: src/providers/fault-count/cat-adi-part2/fault-count.cat-adi-part2.ts (hunk 1 above)
Comment (matthew2564) [reply src/providers/fault-count/cat-adi-part2/fault-count.cat-adi-part2.ts:42]: ok, fair enough - makes sense!

**Review by tomcrawleyy — APPROVED** [review]

**Review by tomcrawleyy — APPROVED** [review]

**Review by matthew2564 — APPROVED** [review]

**Pull request merged**